### PR TITLE
[fix] Item Re-order if Bin does not exists and fixed Stock Projected Qty report to show per Warehouse Re-order level

### PR DESCRIPTION
**Pull request by anandpdoshi**

### CODE REVIEW

File: erpnext/stock/reorder_item.py [hunk 1]
```diff
@@ -15,21 +15,34 @@ def reorder_item():
...
+	warehouse_company = frappe._dict(frappe.db.sql("""select name, company from `tabWarehouse`
+		where disabled=0"""))
 	default_company = (frappe.defaults.get_defaults().get("company") or
 		frappe.db.sql("""select name from tabCompany limit 1""")[0][0])
 
+	items_to_consider = frappe.db.sql_list("""select name from `tabItem` item
+		where is_stock_item=1
+			and (is_purchase_item=1 or is_sub_contracted_item=1)
+			and (end_of_life is null or end_of_life='0000-00-00' or end_of_life > %(today)s)
+			and ((re_order_level and re_order_qty) or exists (
```
Comment (nabinhait) [inline erpnext/stock/reorder_item.py]: re_order_qty is not mandatory

Comment (anandpdoshi) [inline erpnext/stock/reorder_item.py]: What will be taken as re_order_qty if it is not mandatory?

Comment (nabinhait) [inline erpnext/stock/reorder_item.py]: If re_order_qty is not mentioned, request for the shortage qty

File: erpnext/stock/reorder_item.py [hunk 2]
```diff
@@ -15,21 +15,37 @@ def reorder_item():
...
+	warehouse_company = frappe._dict(frappe.db.sql("""select name, company from `tabWarehouse`
+		where disabled=0"""))
 	default_company = (frappe.defaults.get_defaults().get("company") or
 		frappe.db.sql("""select name from tabCompany limit 1""")[0][0])
 
+	items_to_consider = frappe.db.sql_list("""select name from `tabItem` item
+		where is_stock_item=1 and has_variants=0
+			and (is_purchase_item=1 or is_sub_contracted_item=1)
+			and (end_of_life is null or end_of_life='0000-00-00' or end_of_life > %(today)s)
+			and (re_order_level
```
Comment (nabinhait) [inline erpnext/stock/reorder_item.py]: use `(re_order_level is not null and re_order_level != 0)`

**Pull request merged**